File: maxhanna.Server/Controllers/DataContracts/ArrayCharacterItem.cs
namespace maxhanna.Server.Controllers.DataContracts
{
    public class ArrayCharacterItem
    {
        public User? User { get; set; }
        public FileEntry File { get; set; }
        public long Level { get; set; }
        public long Experience { get; set; } 

        public ArrayCharacterItem(User user, FileEntry file, long level, long experience)
        {
            User = user;
            File = file;
            Level = level;
            Experience = experience; 
        } 

    }
}

File: maxhanna.client/src/services/datacontracts/array/array-character-item.ts
 import { FileEntry } from "../file/file-entry";
import { User } from "../user/user";

export class ArrayCharacterItem {
  user: User | undefined;
  file: FileEntry | undefined;
  level: bigint | undefined;
  experience: bigint | undefined;

  constructor(user?: User, file?: FileEntry, level?: bigint, experience?: bigint) {
    this.user = user;
    this.file = file;
    this.level = level;
    this.experience = experience;
  }
}
